include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -231,7 +342,26 @@ class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 			return true;
 		}
 
+		inline size_t getTotalMutableSamplerCount() const { return m_mutableSamplerCount; }
+		inline size_t getTotalDescriptorCount(const E_DESCRIPTOR_TYPE type) const { return m_descriptorCount[type]; }
+
 		core::SRange<const SBinding> getBindings() const { return {m_bindings->data(), m_bindings->data()+m_bindings->size()}; }
```

Comment:
except for `isIdenticallyDefined`, what do we really use this for?

Reply:
`m_bindings` and `m_samplers` are extensively used in `ICPUDescriptorSetLayout`

Reply:
ok I guess more encapsulation needed